runtime/permissions/lib.rs

```diff
@@ -1101,30 +1108,65 @@ impl UnaryPermission<WriteDescriptor> {
     self.check_desc(None, false, api_name, || None)
   }
 }
-
 impl UnaryPermission<NetDescriptor> {
   pub fn query<T: AsRef<str>>(
     &self,
     host: Option<&(T, Option<u16>)>,
   ) -> PermissionState {
-    self.query_desc(
-      host.map(|h| NetDescriptor::new(&h)).as_ref(),
-      AllowPartial::TreatAsPartialGranted,
-    )
+    let net_desc_result = match host {
+      Some(&(ref host_str, port_option)) => {
+        NetDescriptor::new(&&(host_str.as_ref(), port_option))
+      }
+      None => Err(anyhow!("Host information is missing")),
+    };
+    match net_desc_result {
+      Ok(net_desc) => {
+        self.query_desc(Some(&net_desc), AllowPartial::TreatAsPartialGranted)
+      }
+
+      Err(err) => {
+        log::error!("Error creating NetDescriptor: {}", err);
```

Comment:
I was just looking into https://github.com/denoland/deno/issues/23753 and didn't realize this PR fixes the same issue.

However, I think perhaps we should be parsing the host upfront and surfacing the error instead of parsing within here? I did a bit of work on this here (before I saw this PR): https://github.com/denoland/deno/compare/main...dsherret:deno:fix_panic_permissions_fqdn

Reply:
@dsherret 

Hi, Done
please review the changes and let me know if you have any feedback.
https://github.com/denoland/deno/pull/23590/commits/166d1178221219660c84b2e64b7ea232db5ab0e4
Thanks!